scripts/bin/auto-mount

```diff
@@ -1,17 +1,52 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+. "$SNAP/bin/common"
+
+init_mount_dir() {
+    _arg_dir="$1"
+
+}
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                DELETE)   rm_mount  "$_uuid" ;;
+            esac
+        done
+}
+
+try_mount() {
+    _arg_parent="${1%/}"
+    _arg_uuid="$2"
+
+    _source="${_arg_parent}/${_arg_uuid}"
+    _target="${_mount_dir}/${_arg_uuid}"
+
+    log "$SNAP_NAME.auto-mount found: ${_source}"
+
+    if ! mount "${_source}" "${_target}" >/dev/null 2>&1; then
+        log "WARN: Could not mount ${_source} to ${_target}"
+        return
+    fi
+
+    echo "${_target}" >> "${_watch_file}"
+}
+
+main() {
+    set -eu
+
+    # TODO: Should there be a $_log_dir env var in /bin/common?
```

Comment:
I think if we're going to have a `common` and we're putting at least one variable in it, there isn't a strong reason against putting every variable in it (at least the _common_ ones).